pkg/api/v20160330/validate.go

```diff
@@ -71,6 +71,15 @@ func (l *LinuxProfile) Validate() error {
 
 // Validate implements APIObject
 func (a *Properties) Validate() error {
+	if a.OrchestratorProfile == nil {
+		return fmt.Errorf("missing OrchestratorProfile")
+	}
```

Comment:
Semantically I don't think a validation method should do much (any?) transformation. We don't need to bike shed this to death, but is there a better place to do the default property assignments, so that by the time we get to `.Validate()` we're actually doing just that?